packages/core-app-api/src/apis/implementations/auth/pinniped/Pinniped.ts

```diff
@@ -0,0 +1,198 @@
+
+import { OAuth2Response } from '../oauth2/OAuth2';
+import {
+  DefaultAuthConnector,
+  RefreshingAuthSessionManager,
+  SessionManager,
+} from '../../../../lib';
+
+const DEFAULT_PROVIDER = {
+  id: 'pinniped',
+  title: 'Pinniped',
+  icon: () => null,
+};
+
+/**
+ *
+ *
+ * @public
+ */
+export default class Pinniped implements PinnipedSupervisorApi {
+  private audiences: { [aud: string]: SessionManager<OAuth2Session> };
+  private configApi: ConfigApi | undefined;
+  private environment: string;
+  private provider: AuthProviderInfo;
+  private oauthRequestApi: OAuthRequestApi;
+  private discoveryApi: DiscoveryApi;
+  private defaultScopes: string[];
+  private popupOptions: PopupOptions | undefined;
+
+  static create(
+    options: OAuth2CreateOptions,
+  ): typeof pinnipedSupervisorApiRef.T {
+    return new Pinniped(options);
+  }
+
+  private constructor(options: OAuth2CreateOptions) {
+    const {
+      configApi,
+      environment = 'development',
+      provider = DEFAULT_PROVIDER,
+      oauthRequestApi,
+      discoveryApi,
+      defaultScopes = [
+        'openid',
+        'offline_access',
+        'pinniped:request-audience',
+        'username',
+      ],
+      popupOptions,
+    } = options;
+
+    this.configApi = configApi;
+    this.environment = environment;
+    this.provider = provider;
+    this.oauthRequestApi = oauthRequestApi;
+    this.discoveryApi = discoveryApi;
+    this.defaultScopes = defaultScopes;
+    this.popupOptions = popupOptions;
+
+    this.audiences = {};
+  }
+
+  async getSupervisorIdToken(
+    _options?: AuthRequestOptions | undefined,
+  ): Promise<string> {
+    throw new Error('Method not implemented.');
+  }
+
+  async getClusterScopedIdToken(
+    audience: string,
+    options?: AuthRequestOptions | undefined,
+  ): Promise<string> {
+    const aud = await Pinniped.getAudience(audience);
+    if (!(aud in this.audiences)) {
+      const connector = new DefaultAuthConnector({
+        configApi: this.configApi,
+        discoveryApi: this.discoveryApi,
+        environment: this.environment,
+        provider: this.provider,
+        oauthRequestApi: this.oauthRequestApi,
+        sessionTransform({
+          backstageIdentity,
+          ...res
+        }: OAuth2Response): OAuth2Session {
+          const session: OAuth2Session = {
+            ...res,
+            providerInfo: {
+              idToken: res.providerInfo.idToken,
+              accessToken: res.providerInfo.accessToken,
+              scopes: Pinniped.normalizeScopes(
+                scopes => scopes,
+                res.providerInfo.scope,
+              ),
+              expiresAt: res.providerInfo.expiresInSeconds
+                ? new Date(
+                    Date.now() + res.providerInfo.expiresInSeconds * 1000,
+                  )
+                : undefined,
+            },
+          };
+          return session;
+        },
```

Comment:
curious to see what happens if we totally drop this, since we're always requesting the same scopes and aren't interested in supporting user-specified scopes
```suggestion
```